### PR TITLE
log rotate changes

### PR DESCRIPTION
**Pull request by hemalathagajendran**
## Pull request checklist

Log rotate 

No

## Pull request type

 

 

Please check the type of change your PR introduces:
- [ ] Bugfix
- [x] Feature Enhancement
- [ ] Test Automation 
- [ ] Code Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Community Operator listing 
- [ ] Other (please describe): 

## What is the current behavior?

-

## What is the new behavior?
Adding log rotate functionality 
-

## How risky is this change?
- [x] Small, isolated change
- [ ] Medium, requires regression testing
- [ ] Large, requires functional and regression testing

### CODE REVIEW

File: driver/cmd/ibm-spectrum-scale-csi/main.go [hunk 1]
```diff
@@ -177,3 +173,41 @@ func getVerboseLevel(level string) string{
...
+	}
+
+	fileStat, err := logFile.Stat()
+	if err != nil {
+		panic(fmt.Sprintf("failed to stat logger file %v", err))
+	}
+
+	fileStatSize := int(fileStat.Size()) / 1024 / 1024
+
+	// If log file size bigger than rotateSize, will use lunberjack to run the logrotate
```
Comment (Jainbrt) [inline driver/cmd/ibm-spectrum-scale-csi/main.go:199]: type lunberjack -> lumberjack

**Review by Jainbrt — DISMISSED** [review]

File: driver/cmd/ibm-spectrum-scale-csi/main.go (hunk 1 above)
Comment (hemalathagajendran) [reply driver/cmd/ibm-spectrum-scale-csi/main.go]: Fixed typo

**Review by Jainbrt — DISMISSED** [review]

File: driver/cmd/ibm-spectrum-scale-csi/main.go [hunk 2]
```diff
@@ -122,7 +132,7 @@ func setContext() context.Context {
 	return ctx
 }
 
-func getLevel() string{
+func getLevel() string {
 	level := os.Getenv(logLevel)
 	var logValue string
 	klog.Infof("logValue: %s", level)
```
Comment (deeghuge) [inline driver/cmd/ibm-spectrum-scale-csi/main.go:138]: if env is not set log will show empty value. Better to print after the env check

File: driver/cmd/ibm-spectrum-scale-csi/main.go [hunk 3]
```diff
@@ -177,3 +173,41 @@ func getVerboseLevel(level string) string{
 	}
 }
 
+func InitFileLogger() func() {
+	filePath := hostPath + dirPath + "/" + logFile
+	_, err := os.Stat(filePath)
+	if os.IsNotExist(err) {
+		fileDir, _ := path.Split(filePath)
+		err := os.MkdirAll(fileDir, 0766)
```
Comment (deeghuge) [inline driver/cmd/ibm-spectrum-scale-csi/main.go:181]: please change permission to 755

Comment (hemalathagajendran) [reply driver/cmd/ibm-spectrum-scale-csi/main.go]: Modified permission

File: driver/cmd/ibm-spectrum-scale-csi/main.go (hunk 2 above)
Comment (hemalathagajendran) [reply driver/cmd/ibm-spectrum-scale-csi/main.go]: Modified accordingly

**Review by amdabhad — APPROVED** [review]

**Review by deeghuge — APPROVED** [review]

**Pull request merged**